src/components/auth/forgotPassword/forgotPasswordForm.component.tsx

```diff
@@ -8,83 +8,73 @@ import {
   ThemeType,
   withStyles,
 } from '@kitten/theme';
-import {
-  Button,
-  ValidationInput,
-} from '@src/components/common';
+import { ValidationInput } from '@src/components/common';
 import { EmailIconFill } from '@src/assets/icons';
+import { EmailValidator } from '@src/core/validators';
 
 export interface ForgotPasswordFormType {
   email: string;
 }
 
 interface ComponentProps {
-  onSubmit: (value: ForgotPasswordFormType) => void;
+  /**
+   * Will emit changes depending on validation:
+   * Will be called with form value if it is valid, otherwise will be called with undefined
+   */
+  onFormValueChange: (value: ForgotPasswordFormType | undefined) => void;
 }
 
+
 export type ForgotPasswordFormProps = ThemedComponentProps & ViewProps & ComponentProps;
 
 interface State {
-  email: string;
-  emailValid: boolean;
+  email: string | undefined;
 }
 
 class ForgotPasswordFormComponent extends React.Component<ForgotPasswordFormProps, State> {
 
   public state: State = {
-    email: '',
-    emailValid: false,
+    email: undefined,
   };
 
-  private emailValidator: RegExp = /\S+@\S+\.\S+/;
+  public componentDidUpdate(prevProps: ForgotPasswordFormProps, prevState: State) {
+    const oldFormValid: boolean = this.isValid(prevState);
+    const newFormValid: boolean = this.isValid(this.state);
 
-  private onSubmitButtonPress = () => {
-    const { email } = this.state;
+    const becomeValid: boolean = !oldFormValid && newFormValid;
+    const becomeInvalid: boolean = oldFormValid && !newFormValid;
 
-    this.props.onSubmit({ email });
-  };
+    if (becomeValid) {
+      this.props.onFormValueChange(this.state);
+    } else if (becomeInvalid) {
+      this.props.onFormValueChange(undefined);
+    }
+  }
 
-  private onEmailValidationResult = (emailValid: boolean, email: string) => {
-    this.setState({
-      emailValid,
-      email,
-    });
+  private onEmailInputTextChange = (email: string) => {
+    this.setState({ email });
   };
 
-  private isFormValid = (): boolean => {
-    const { emailValid } = this.state;
+  private isValid = (value: ForgotPasswordFormType): boolean => {
+    const { email } = value;
 
-    return emailValid;
+    return email !== undefined;
```

Comment:
Looks strange.

Reply:
`ValidationInput` emits `onChangeText` when value is valid with it's value, with `undefined` otherwise